lib/post_request_task/task.py

```diff
@@ -0,0 +1,113 @@
+import threading
+from functools import partial
+
+from django.core.signals import (got_request_exception, request_finished,
+                                 request_started)
+
+import commonware.log
+from celery import task as base_task
+from celery import Task
+
+
+log = commonware.log.getLogger('z.post_request_task')
+
+
+_locals = threading.local()
+
+
+def _get_task_queue():
+    """Returns the calling thread's task queue."""
+    return _locals.__dict__.setdefault('task_queue', [])
+
+
+def _start_queuing_tasks(**kwargs):
+    """Starts queuing tasks for this thread.
+
+    Not supposed to be called directly, instead it's connected to the
+    request_started signal.
+
+    If not called, tasks are delayed normally (so tasks still function without
+    having to call _send_tasks_and_stop_queuing() manually when we're outside
+    the request-response cycle.."""
+    _locals.__dict__['queue_tasks'] = True
+
+
+def _stop_queuing_tasks(**kwargs):
+    """Stops queuing tasks for this thread.
+
+    Not supposed to be called directly, only useful for tests cleanup."""
+    _locals.__dict__.pop('queue_tasks', None)
```

Comment:
this seems weird to me that it would completely remove the `queue_tasks` key. Why not just set it to False?